build/prow/e2e/Dockerfile

```diff
@@ -69,15 +85,9 @@ ENTRYPOINT ["/usr/bin/bash"]
 FROM gke-e2e as kind-e2e
 
 WORKDIR /repo
-# Steps after here can't be cached since they touch the local filesystem.
-# Just copy everything in the nomos repository so we have whatever we might need.
-COPY . .
-
-# install the nomos CLI, which is used in several e2e tests.
-RUN go install ./cmd/nomos
```

Comment:
The `nomos` cli is built in this layer because the `kind-e2e` image is special in that is rebuilt for every job execution (using docker-in-docker). As such, `kind-e2e` is always built from the current commit. We always want to use the current version of the nomos binary for testing, so it needs to be created during the job execution.

For gke-e2e periodics, it's pulled from postsubmit artifacts in the gke-e2e container as part of the job entrypoint (make target). The gke-e2e image itself shouldn't have the `nomos` binary pre-installed.

For kind-e2e presubmits, it's built as part of the kind-e2e image build on every job execution.

Reply:
Don't 3/4 layers get re-built every time, since the source is copied into the builder?

I removed the `COPY . .` from the kind-e2e layer because it seemed redundant and expensive. I replaced it with a mount of the repo in the docker run call in Makefile.e2e. Previously, the docker build was copying like 4GiB of context data every time it built, probably due to the .git & .output dirs, which was hella slow.

The way you're describing how it works makes me think we're conflating the target environment and whether it's presubmit/postsumbmit. We also run kind in the postsubmits now, and may want to use gke in presubmits in the future. So maybe we should move this nomos build elsewhere?

Reply:
Agreed that `gke-e2e` should remain separate from the source, and only gets changed when the golang base image needs to be updated. For periodic jobs, `nomos` CLI is installed as a runtime make target, `__install-nomos-local`.

Currently, the `builder` image has a copy of the source, which seems unnecessary. I left a comment above to remove that layer.